tests/MongoDB.Driver.Tests/PinnedShardRouterTests.cs

```diff
@@ -172,11 +172,17 @@ private DisposableMongoClient CreateDisposableClient(EventCapturer eventCapturer
             var client = DriverTestConfiguration.CreateDisposableClient((MongoClientSettings settings) =>
                 {
                     settings.ClusterConfigurator = c => c.Subscribe(eventCapturer);
-                    settings.LocalThreshold = TimeSpan.FromMilliseconds(1000);
+                    settings.LocalThreshold = TimeSpan.FromMilliseconds(5000);
                 }, 
                 useMultipleShardRouters);
             var timeOut = TimeSpan.FromSeconds(60);
-            SpinWait.SpinUntil(() => client.Cluster.Description.Type != ClusterType.Unknown, timeOut).Should().BeTrue();
+
+            var cluster = client.Cluster;
+            SpinWait.SpinUntil(condition: () =>
+                cluster.Description.Type != ClusterType.Unknown
+                && cluster.Description.Servers.Count >= (useMultipleShardRouters ? 2 : 1)
```

Comment:
Note: `cluster.Description.Servers.Count` is a constant (based on how many servers where listed on the connection string).

Reply:
Ah oops, thanks for the clarification!